rustls/src/msgs/message.rs

```diff
@@ -3,17 +3,24 @@ use crate::enums::{AlertDescription, ContentType, HandshakeType};
 use crate::error::{Error, InvalidMessage, PeerMisbehaved};
 use crate::internal::record_layer::RecordLayer;
 use crate::msgs::alert::AlertMessagePayload;
-use crate::msgs::base::Payload;
+use crate::msgs::base::{BorrowedPayload, Payload};
 use crate::msgs::ccs::ChangeCipherSpecPayload;
-use crate::msgs::codec::{Codec, Reader};
+use crate::msgs::codec::{Codec, Reader, ReaderMut};
 use crate::msgs::enums::AlertLevel;
 use crate::msgs::fragmenter::MAX_FRAGMENT_LEN;
 use crate::msgs::handshake::HandshakeMessagePayload;
 
 use alloc::vec::Vec;
 
-use super::base::BorrowedPayload;
-use super::codec::ReaderMut;
+#[derive(Debug)]
+pub enum MessageError {
+    TooShortForHeader,
+    TooShortForLength,
+    InvalidEmptyPayload,
+    MessageTooLarge,
+    InvalidContentType,
+    UnknownProtocolVersion,
+}
```

Comment:
I think we want this `enum` to be located between `BorrowedOpaqueMessage` and `read_opaque_message_header` to match the [top-down ordering](https://github.com/rustls/rustls/blob/main/CONTRIBUTING.md#top-down-ordering-within-modules) convention

It should be after `BorrowedOpaqueMessage` and `OpaqueMessage` because those types rely on it. It should be before `read_opaque_message_header` because the error type is `pub` and that helper isn't.